src/components/GalleryTariffs/GalleryTariffs.tsx

```diff
@@ -0,0 +1,117 @@
+import { BlockWrapper } from '../../ui-kit/BlockWrapper';
+import { FoldableSection } from '../../ui-kit/Foldable/FoldableSection';
+import { Foldable } from '../../ui-kit/Foldable/Foldable';
+import { FoldablePartProps } from '../../ui-kit/Foldable/FoldablePartProps';
+import { IconName } from '../../icons/IconName';
+import { VNode } from '../../model/VNode';
+import type { UniBlockProps } from '../../model/JSXBlock';
+import type { HeadlineCommonProps } from '../../model/HeadlineType';
+import type { GalleryTariffsContent, TariffsCardCommonProps } from './GalleryTariffsContent';
+
+export interface GalleryTariffsProps extends GalleryTariffsContent, UniBlockProps {}
+
+export const GalleryTariffs = JSX<GalleryTariffsProps>(
+  ({ context, className = '', title, description, cards, ...rest }) => (
+    <BlockWrapper
+      className={`font-sans bg-white p-[50px] ${className}`}
+      context={context}
+      {...rest}
+    >
+      <Headline
+        context={context}
+        className="!p-0 mb-10"
+        title={title}
+        description={description}
+        headlineVersion="M"
+        align="center"
+      />
+      {cards?.length ? <div className="flex">{cards.map(renderCard)}</div> : null}
+    </BlockWrapper>
+  ),
+);
+
+const renderCard = (card: TariffsCardCommonProps, i: number): VNode => {
+  const hitStyle = getHitStyle(Boolean(card?.isHighlighting));
+  const rows = (card?.rows || []).map(renderCardInfo);
+  const [visibleRows, hiddenRows] = [rows.slice(0, 1), rows.slice(1)];
+
+  return (
+    <div
+      key={String(i)}
+      className={`p-8 border border-main-stroke relative w-[322px] -ml-[1px] ${hitStyle}`}
+    >
+      {card?.isHighlighting ? renderHitIcon() : null}
+      {card?.icon?.src ? <Img className="flex justify-center" image={card?.icon} /> : null}
+      {card?.title ? (
+        <Heading
+          className="text-primary-text text-center pt-2"
+          headingType="h5"
+          title={card.title}
+        />
+      ) : null}
+      {card?.button ? <Button className="w-full mt-4" {...card.button} /> : null}
+      {card?.rows?.length ? (
+        <Foldable
+          renderFoldableSection={({ isUnfolded }) => (
+            <div>
+              {visibleRows}
+              <FoldableSection isUnfolded={isUnfolded}>{hiddenRows}</FoldableSection>
+            </div>
+          )}
+          renderFoldButton={renderFoldButton}
+        />
+      ) : null}
+    </div>
+  );
+};
+
+const getHitStyle = (isHighlighting: boolean): string => (isHighlighting ? 'pt-[70px]' : 'mt-9');
+
+const renderFoldButton = ({ isUnfolded, onToggle }: FoldablePartProps): VNode => {
+  const labels = ['Подробнее о тарифе', 'Скрыть'];
+  const icons: IconName[] = ['ArrowDownIcon', 'ArrowUpIcon'];
+
+  return (
+    <button
+      className="w-full cursor-pointer text-primary-main pt-5 flex justify-between items-center"
+      onClick={onToggle}
+    >
+      <div>{labels[Number(isUnfolded)]}</div>
```

Comment:
У функции вроде вообще нулевая complexity, мб лучше:
```suggestion
      <div>{isUnfolded ? 'Скрыть' : 'Подробнее о тарифе'}</div>
```

Тогда можно будет и return убрать

Reply:
Done